lib/sparrow/pools_warden.ex

```diff
@@ -108,7 +134,15 @@ defmodule Sparrow.PoolsWarden do
         pid=#{inspect(pid)}, reason=#{inspect(reason)}"
       end)
 
-    {:noreply, Map.delete(state, pid)}
+    new_state = Map.delete(state, pid)
+
+    :telemetry.execute(
+      [:sparrow, :pools_warden, :pool_down],
+      %{},
+      %{reason: reason}
```

Comment:
it would be great to add `pool_type`, `pool_name`, `pool_tags` metadata here and also in the add pool.